File: RetroGameGauntlet.Droid/Adapters/AppInfoAdapter.cs
using System;
using Android.App;
using Android.Content.PM;
using Android.Util;
using RetroGameGauntlet.PCL.Adapters;

namespace RetroGameGauntlet.Droid.Adapters
{
    public class AppInfoAdapter : IAppInfoAdapter
    {
        private const string Tag = "AppInfoAdapter";

        public AppInfoAdapter()
        {
        }

        public string VersionName
        {
            get
            {
                try
                {
                    return Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName, 
                                                                             PackageInfoFlags.MetaData).VersionName;
                }
                catch (Exception e)
                {
                    //We cannot log it using MetroLog (will cause StackOverflow crash)
                    Log.Error(Tag, "Cannot get app version", e);
                    return string.Empty;
                }
            }
        }

        public string VersionCode
        {
            get
            {
                try
                {
                    return Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName,
                                                                             PackageInfoFlags.MetaData).VersionCode.ToString();
                }
                catch (Exception e)
                {
                    //We cannot log it using MetroLog (will cause StackOverflow crash)
                    Log.Error(Tag, "Cannot get app version code", e);
                    return "0";
                }
            }
        }
    }
}

File: RetroGameGauntlet.Droid/Effects/AppInfoAdapter.fs
namespace RetroGameGauntlet.Droid.Adapters

open System
open Android.App
open Android.Content.PM
open RetroGameGauntlet.Core.Adapters

type AppInfoAdapter () =
    interface IAppInfoAdapter with
        member this.VersionName = Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName, 
                                                                                     PackageInfoFlags.MetaData).VersionName;

        member this.VersionCode = Application.Context.PackageManager.GetPackageInfo(Application.Context.PackageName,
                                                                                     PackageInfoFlags.MetaData).VersionCode.ToString();